### PR TITLE
Add upgrade test

### PR DESCRIPTION
**Pull request by cpanato**
- default mattermost-operator to use mattermost version 5.11.0
- add upgrade test

### CODE REVIEW

File: test/e2e/mattermost_test.go [hunk 1]
```diff
@@ -90,9 +94,94 @@ func mattermostScaleTest(t *testing.T, f *framework.Framework, ctx *framework.Te
...
+	assert.Nil(t, err)
+
 	return nil
 }
 
+func mattermostUpgradeTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	assert.Nil(t, err)
+
+	// create memcached custom resource
```
Comment (gabrieljackson) [inline test/e2e/mattermost_test.go]:
```suggestion
	// create clusterInstallation custom resource
```

File: test/e2e/mattermost_test.go [hunk 2]
```diff
@@ -90,9 +94,94 @@ func mattermostScaleTest(t *testing.T, f *framework.Framework, ctx *framework.Te
...
+
+	// Wait for this pod be terminated
+	err = e2eutil.WaitForDeletion(t, f.Client.Client, mmOldPod, retryInterval, timeout)
+	assert.Nil(t, err)
+
+	// wait for deployment be completed
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "test-mm2", 1, retryInterval, timeout)
+	assert.Nil(t, err)
+
+	time.Sleep(time.Second * 5)
```
Comment (gabrieljackson) [inline test/e2e/mattermost_test.go]: Do we need this sleep? The very next function seems to wait anyway.

File: test/e2e/utils.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func waitForReconcilicationComplete(t *testing.T, dynclient client.Client, namespace, name string, retryInterval, timeout time.Duration) error {
+	newMattermost := &operator.ClusterInstallation{}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		errClient := dynclient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, newMattermost)
+		if errClient != nil {
+			return false, err
```
Comment (gabrieljackson) [inline test/e2e/utils.go]:
```suggestion
			return false, errClient
```

File: test/e2e/utils.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		}
+		t.Logf("Waiting for Reconcilication finish (Status:%s)\n", newMattermost.Status.State)
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+	t.Logf("Reconcilication completed (%s)\n", newMattermost.Status.State)
+	return nil
+
```
Comment (gabrieljackson) [inline test/e2e/utils.go]: Extra newline

File: test/e2e/mattermost_test.go (hunk 2 above)
Comment (cpanato) [reply test/e2e/mattermost_test.go]: maybe not, removed

File: test/e2e/utils.go (hunk 3 above)
Comment (cpanato) [reply test/e2e/utils.go]: thanks!

File: test/e2e/utils.go (hunk 4 above)
Comment (cpanato) [reply test/e2e/utils.go]: all done @gabrieljackson 
thanks for your review!

**Pull request merged**